### PR TITLE
chore(recipes): bump kai-scheduler v0.14.1 and kubeflow-trainer 2.2.0

### PR DESCRIPTION
**Pull request by yuanchen8911**
## Summary

Two Phase-2 follow-ups from #698, batched into one PR because both are small chart-pin changes coupled to a single non-pin tweak each. Also includes minor docs cleanup picked up alongside.

| Component | Current | New | Coupled change |
|---|---|---|---|
| `kai-scheduler` | `v0.13.0` | `v0.14.1` | OCI registry namespace migration (`oci://ghcr.io/nvidia/kai-scheduler` → `oci://ghcr.io/kai-scheduler/kai-scheduler`) |
| `kubeflow-trainer` | `2.1.0` | `2.2.0` | Validator fallback archive URL bump in `validators/performance/trainer_lifecycle.go` |

## Motivation / Context

Both bumps were excluded from the Phase-1 PR (#715) for clean reasons:

- **kai-scheduler** — when the upstream repo was transferred from `NVIDIA/` to `kai-scheduler/` org, chart publishing moved with it. The old `ghcr.io/nvidia/kai-scheduler` namespace is frozen at `v0.13.0`; the full release stream lives at `ghcr.io/kai-scheduler/kai-scheduler`. AICR's recipe was still pointing at the frozen old path. This is an OCI source migration *plus* a version bump — coupled changes that belong together.
- **kubeflow-trainer** — the chart pin in `recipes/registry.yaml` is coupled with the hardcoded fallback archive URL in `validators/performance/trainer_lifecycle.go`. The validator's no-CRD install path downloads a hardcoded `v2.1.0` GitHub archive; if we bump the chart pin without bumping the URL, the fallback installs `v2.1.0` manifests while the chart deploys `v2.2.0`. To keep #715 pure config / no Go changes, this was deferred.

### kai-scheduler — verified clean

- 41/41 rendered templates and identical kinds/counts vs `v0.13.0`
- Only `values.yaml` addition is an opt-in `vpa:` block (`enabled: false` default)
- Our customizations (`global.tolerations`, `admission.gpuPodRuntimeClassName`, `postCleanup.enabled`) all still apply unchanged
- New OCI namespace verified pullable: `helm pull oci://ghcr.io/kai-scheduler/kai-scheduler/kai-scheduler --version v0.14.1` succeeds

### kubeflow-trainer — verified clean

- v2.2.0 archive layout unchanged from v2.1.0: same `manifests/overlays/manager` kustomize, same `trainjobs.trainer.kubeflow.org/v1alpha1` CRD identity, same `kubeflow-system` namespace
- Only difference upstream is the controller-manager image tag
- Chart `oci://ghcr.io/kubeflow/charts/kubeflow-trainer --version 2.2.0` verified pullable

### Companion fixes

- **`examples/recipes/aks-training.yaml`** — refresh the kai-scheduler example pin (source URL + version) to track the new registry default. Matches the example-pin convention from #715 (#283, #336, #450) — only this one example references kai-scheduler.
- **`docs/user/component-catalog.md`** — update the KAI Scheduler upstream link from `github.com/NVIDIA/KAI-Scheduler` to the new `github.com/kai-scheduler/KAI-Scheduler` (the upstream GitHub repo migrated alongside the OCI registry).
- **`docs/contributor/data.md` and `docs/contributor/component.md`** — update illustrative cert-manager YAML/JSON snippets from `v1.17.2` → `v1.20.2` to match the post-#715 registry default. These were tagged as cosmetic-drift in the #715 cross-review and deferred; rolling them in here.

Fixes: N/A
Related: #698 (follow-up items 3 and 5), follows-up #715 (Phase-1 PR)

## Type of Change

- [x] Build/CI/tooling
- [x] Documentation update

## Component(s) Affected

- [x] Recipe engine / data (`pkg/recipe`) — registry default + base overlay pins
- [x] Validator (`pkg/validator`) — fallback archive URL in `validators/performance/trainer_lifecycle.go`
- [x] Docs/examples (`docs/`, `examples/`) — example pin refresh, KAI Scheduler upstream link, contributor doc snippet versions

## Implementation Notes

- The kai-scheduler change is two coupled lines per file (registry default + version, then overlay source + version) — the OCI source path and version pin must move together.
- The kubeflow-trainer change is one registry version line plus three lines in the Go validator (URL constant + 2 doc comments referencing the version). The validator's behavior is otherwise unchanged: same kustomize overlay path, same CRD identity check.
- **Two v2.2.0 breaking-change consequences addressed in this PR:**
  - `PodTemplateOverrides` is replaced by `runtimePatches` ([kubeflow/trainer#3309](https://github.com/kubeflow/trainer/pull/3309)). The CRD still admits the old field name for compat, but the v2.2 controller no longer applies it; pods come out with no override fields. The `pytorch-mnist` demo TrainJob in `demos/cuj1-eks.md` and `demos/cuj1-gke.md` is migrated to the `runtimePatches` shape with `manager: aicr.nvidia.com/demo` and explicit per-cluster scheduling (EKS demo: `dedicated=worker-workload:NoSchedule|NoExecute`; GKE demo: `dedicated=gpu-workload:NoSchedule` + `nvidia.com/gpu=present:NoSchedule` to match the rest of the GKE flow).
  - `mlPolicy.torch.numProcPerNode` is removed ([kubeflow/trainer#3239](https://github.com/kubeflow/trainer/pull/3239)) — Torch now infers parallelism from the container's `nvidia.com/gpu` resource limit. AICR's `torch-distributed` ClusterTrainingRuntime in `recipes/components/kubeflow-trainer/manifests/` is updated from `mlPolicy.torch: { numProcPerNode: auto }` to `mlPolicy.torch: {}`, matching the v2.2.0 reference runtime. `mlPolicy.mpi.numProcPerNode` is unaffected upstream, so MPI test fixtures stay as-is.
- Net diff: 13 files, +75/-34 lines. The `site/docs/` mirror is gitignored (auto-generated from `docs/`); only the canonical `docs/` was edited.

## Known caveat 1 — kubeflow-trainer v2.1.0 -> v2.2.0 upgrades leave CRDs stale

Helm 3/4 does NOT upgrade CRDs on `helm upgrade` — only on first install (CRDs ship in the chart's `crds/` directory, which Helm treats as install-only). Clusters that previously deployed kubeflow-trainer v2.1.0 retain the v2.1.0 CRD after `helm upgrade` to v2.2.0. The v2.1.0 CRD has `podTemplateOverrides` but no `runtimePatches` — so the migrated demo TrainJob is rejected at admission with `unknown field "spec.runtimePatches"` until CRDs are explicitly upgraded:

```bash
helm pull oci://ghcr.io/kubeflow/charts/kubeflow-trainer --version 2.2.0 --untar -d /tmp/
kubectl apply -f /tmp/kubeflow-trainer/crds/ --server-side --force-conflicts
```

Fresh deploys (via `aicr bundle` + `deploy.sh`) are unaffected — Helm installs the v2.2.0 CRDs at first install.

A follow-up improvement is to have the bundler emit CRD upgrade commands in `install.sh` for charts that ship `crds/`, but that's out of scope for this version-bump PR.

## Known caveat 2 — kai-scheduler default queue requires explicit priorityClass for Deployment-style workloads

The kai-scheduler chart (both v0.13.0 and v0.14.1) ships `default-queue` and `default-parent-queue` with `gpu.quota: 0` and `limit: -1`. Combined with kai's pod-grouper priority-class auto-assignment, this means:

- **Bare Pods / Job-style workloads** → pod-grouper assigns `priorityClassName: train` (priority 50, **preemptible**) → can go over `quota: 0` because `limit: -1` is unlimited. Examples: the `gang-scheduling-test.yaml` in `pkg/evidence/scripts/manifests/` and TrainJob-driven workloads. **These work out of the box.**
- **Deployment / ReplicaSet workloads** → pod-grouper's Deployment Grouper auto-assigns `priorityClassName: inference` (priority 125, **non-preemptible**) → blocked by `quota: 0` with `NonPreemptibleOverQuota`. **These fail until either the workload sets an explicit preemptible priorityClass or the queue's quota is raised.**

Verified empirically on a deployed cluster (kai v0.14.1):
- `gang-scheduling-test.yaml` applied as-is on a fresh cluster → both pods scheduled, completed in ~14s (auto-`train` priority).
- 2-replica Deployment with `schedulerName: kai-scheduler` and DRA `ResourceClaimTemplate` requesting `gpu.nvidia.com` devices → blocked with `NonPreemptibleOverQuota` (auto-`inference` priority).
- Same Deployment with explicit `priorityClassName: train` added to the pod template → both replicas scheduled, each with its own H100 via DRA. Quota stayed at 0; no chart values change required.

Workaround for Deployment-style kai workloads: set `priorityClassName: train` (or `build` / `build-preemptible`, depending on workload semantics) on the pod template. The chart ships these priorityClasses in `templates/priorityclasses/`.

This is **pre-existing kai/AICR behavior** — the v0.13.0 chart shipped the same defaults; it's surfaced now because PR #720 is the first time the kai-scheduler bump is being explicitly tested with a `schedulerName: kai-scheduler` Deployment-style workload. A follow-up improvement could either (a) document this prominently in `docs/user/...` or (b) override `defaultQueue.parentResources.gpu.quota` in `recipes/components/kai-scheduler/values.yaml` so inference-style workloads work without per-workload priorityClass tagging. Out of scope for this version-bump PR.

## Testing

```bash
make tidy                                   # no-op (clean)
make lint                                   # 0 issues
go test -count=1 ./pkg/recipe/...           # ok
go test -count=1 ./validators/performance/... # ok

# End-to-end: bundle a kubeflow-using EKS training recipe
$ aicr recipe --service eks --accelerator h100 --intent training \
              --os ubuntu --platform kubeflow -o recipe.yaml
$ aicr bundle -r recipe.yaml -o /tmp/bundle
... succeeds; kai-scheduler and kubeflow-trainer per-component
    install.sh artifacts produced cleanly.
```

## Risk Assessment

- [x] **Low** — Both bumps are minor within the same major; values surfaces are unchanged (kai-scheduler) or cosmetic-only (kubeflow-trainer); the OCI namespace migration is a registry-side path change verified end-to-end. Doc changes are illustrative / link updates only.

**Rollout notes:** No migration steps. Bundles regenerated post-merge will pull from the new kai-scheduler OCI namespace and the new kubeflow-trainer chart version. Existing installations are unaffected until re-bundled. The validator fallback path will install the v2.2.0 trainer if invoked on a cluster without the chart pre-installed.

## Checklist

- [x] Tests pass locally (`make test` with `-race`)
- [x] Linter passes (`make lint`)
- [x] I did not skip/disable tests to make CI green
- [x] I added/updated tests for new functionality (N/A — version bumps + URL constant)
- [x] I updated docs if user-facing behavior changed (KAI link refresh + cert-manager doc snippet versions)
- [x] Changes follow existing patterns in the codebase
- [x] Commits are cryptographically signed (`git commit -S`)

### CODE REVIEW

**Comment by yuanchen8911** [comment]
@coderabbitai review

**Review by mchmarny — DISMISSED** [review]

File: recipes/registry.yaml [hunk 1]
```diff
@@ -359,9 +359,9 @@ components:
     healthCheck:
       assertFile: checks/kai-scheduler/health-check.yaml
     helm:
-      defaultRepository: oci://ghcr.io/nvidia/kai-scheduler
+      defaultRepository: oci://ghcr.io/kai-scheduler/kai-scheduler
       defaultChart: kai-scheduler
-      defaultVersion: v0.13.0
+      defaultVersion: v0.14.1
```
[the remark this answers is not in the document]
Comment (yuanchen8911) [reply recipes/registry.yaml:369]: Applied as suggested, plus the matching golden-file updates so the fixture stays self-consistent:

- `pkg/bundler/deployer/helm/helm_test.go:1898-1899` — `Version: "v0.13.0" → "v0.14.1"`, `Source: "oci://ghcr.io/nvidia/kai-scheduler" → "oci://ghcr.io/kai-scheduler/kai-scheduler"`
- `pkg/bundler/deployer/helm/testdata/kai_scheduler_present/README.md` — catalog row updated to `v0.14.1` and the new OCI namespace
- `pkg/bundler/deployer/helm/testdata/kai_scheduler_present/001-kai-scheduler/upstream.env` — `CHART` URL and `VERSION` updated

`go test ./pkg/bundler/deployer/helm/...` passes. Force-pushed as `7b0951e4` (rebased onto latest `main`).

**Comment by yuanchen8911** [comment]
@coderabbitai review

**Review by lockwobr — APPROVED** [review]
